plugins/GigPlayer/GigPlayer.cpp

```diff
@@ -440,50 +430,44 @@ void GigInstrument::play( SampleFrame* _working_buffer )
 				samples = frames / freq_factor + Sample::s_interpolationMargins[m_interpolation];
 			}
 
+			const auto neededCapacity = static_cast<std::vector<SampleFrame>::size_type>(samples);
+			sampleData.resize(std::max(neededCapacity, sampleData.capacity()));
+			convertBuf.resize(std::max(neededCapacity, convertBuf.capacity()));
+
 			// Load this note's data
-			SampleFrame sampleData[samples];
 			loadSample(sample, sampleData, samples);
 
 			// Apply ADSR using a copy so if we don't use these samples when
 			// resampling, the ADSR doesn't get messed up
 			ADSR copy = sample.adsr;
-
-			for( f_cnt_t i = 0; i < samples; ++i )
-			{
-				float amplitude = copy.value();
+			const auto amplitude = copy.value();
+#if ((_MSC_VER >= 1914) || (__GNUC__ >= 9 && __GNUC_MINOR__ >= 3) || (__clang__ && __clang_major__ >= 5))
+			std::for_each_n(sampleData.begin(), samples, [&](auto& frame) {
+				frame[0] *= amplitude;
+				frame[1] *= amplitude;
+			});
+#else
+			// TODO: Ubuntu 20.04 CI change should be enough to remove this if/else
```

Comment:
We use Ubuntu 20.04 for the Linux and MinGW builds now, so maybe this can be removed

Reply:
done